modules/frontend/traceid_sharder.go

```diff
@@ -40,13 +39,11 @@ func newAsyncTraceIDSharder(cfg *TraceByIDConfig, logger log.Logger) pipeline.As
 
 // RoundTrip implements http.RoundTripper
 func (s asyncTraceSharder) RoundTrip(pipelineRequest pipeline.Request) (pipeline.Responses[combiner.PipelineResponse], error) {
-	r := pipelineRequest.HTTPRequest()
-
-	ctx, span := tracer.Start(r.Context(), "frontend.ShardQuery")
+	ctx, span := tracer.Start(pipelineRequest.Context(), "frontend.ShardQuery")
 	defer span.End()
-	r = r.WithContext(ctx)
+	pipelineRequest.WithContext(ctx)
```

Comment:
Not sure if this creates more allocation than just passing the context to the clone function